feature_engine/transformation/date_transformer.py

```diff
@@ -0,0 +1,49 @@
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.variable_manipulation import _check_input_parameter_variables
+
+
+class date_feature_extractor(BaseNumericalTransformer):
+    def __init__(
+        self,
+        features: Union[None, int, str, List[Union[str, int]]] = None,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+    ) -> None:
+
+        if not fearures:
```

Comment:
mind the typo all throughout the code, should be features not fearures. Also, please change name as indicated above.

This error should check that if the user passes a value different from None, the value is one of:
  "year",  "semester", "quarter", "month", "week",  "day", "day_of_week", "is_weekend", "hr", "min", "sec".

The trasnformer also should be able to return all of those features. The problem that we face is, to discriminate if the datetime variable has time and date component or only 1 of the 2.